include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h

```diff
@@ -161,38 +161,66 @@ class __reduction_pack
     }
 };
 
-// Sequenced version of for_loop_n
-template <typename _ExecutionPolicy, typename _Ip, typename _Size, typename _Function, typename... _Rest>
+// Sequenced + vectorized version of for_loop_n
+template <class _Tag, typename _ExecutionPolicy, typename _Ip, typename _Size, typename _Function, typename... _Rest>
 void
-__pattern_for_loop_n(_ExecutionPolicy&&, _Ip __first, _Size __n, _Function __f, __single_stride_type,
-                     /*vector=*/::std::false_type, /*parallel=*/::std::false_type, _Rest&&... __rest) noexcept
+__pattern_for_loop_n(_Tag, _ExecutionPolicy&&, _Ip __first, _Size __n, _Function __f, __single_stride_type,
+                     _Rest&&... __rest) noexcept
 {
+    static_assert(__is_backend_tag_serial_v<::std::decay_t<_Tag>> ||
+                  __is_backend_tag_parallel_forward_v<::std::decay_t<_Tag>>);
+
     __reduction_pack<_Rest...> __pack{__reduction_pack_tag(), ::std::forward<_Rest>(__rest)...};
 
-    for (_Size __i = 0; __i < __n; ++__i, ++__first)
-        __pack.__apply_func(__f, __first, __i);
+    if constexpr (typename _Tag::__is_vector{})
+    {
+        oneapi::dpl::__internal::__brick_walk1(
```

Comment:
Yes, for this place it's true, because the changes in this file is bigger then usually: there were a lot of functions and specializations in the previous code state. Now when we using tags we combine some specializations together.